### PR TITLE
Bug 2092047: cncc: add RBAC coordination.k8s.io leases

### PR DESCRIPTION
**Pull request by flavio-fernandes**
Requirement for https://github.com/openshift/cloud-network-config-controller/pull/44
so it can rebase to use k8s.io/api v0.24.0

Signed-off-by: Flavio Fernandes <flaviof@redhat.com>

### CODE REVIEW

**Comment by flavio-fernandes** [comment]
@andreaskaris @ricky-rav @kyrtapz PTAL

**Comment by zshi-redhat** [comment]
@flavio-fernandes  We would need a separate PR to add the same role to hypershift cno deployment here: https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go#L131, so that cno can create the roles in hypershift management cluster.

**Comment by andreaskaris** [comment]
/lgtm 

Fingers crossed that those are enough rights (I'm a bit concerned that we have create, but not leased). But I'm assuming that you tested it with that and that it's good.

**Comment by kyrtapz** [comment]
> @flavio-fernandes We would need a separate PR to add the same role to hypershift cno deployment here: https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go#L131, so that cno can create the roles in hypershift management cluster.

I don't think it applies to hypershift. The change is meant for CNCC which, in hypershift, is deployed only in the hosted cluster .

**Comment by kyrtapz** [comment]
> @flavio-fernandes We would need a separate PR to add the same role to hypershift cno deployment here: https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go#L131, so that cno can create the roles in hypershift management cluster.

I don't think it applies to hypershift. The change is meant for CNCC which, in hypershift, is deployed only in the hosted cluster .

**Comment by zshi-redhat** [comment]
> > @flavio-fernandes We would need a separate PR to add the same role to hypershift cno deployment here: https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go#L131, so that cno can create the roles in hypershift management cluster.
> 
> I don't think it applies to hypershift. The change is meant for CNCC which, in hypershift, is deployed only in the hosted cluster .

oh, you're right. I mixed the slack thread discussion and this PR, I was meant to say this is needed when the ovnk rebase patches are downstreamed: https://github.com/ovn-org/ovn-kubernetes/pull/3001/commits/d2cd0ba41da59fb2da3202dbb16a165c0e46baca

**Comment by flavio-fernandes** [comment]
> /lgtm
> 
> Fingers crossed that those are enough rights (I'm a bit concerned that we have create, but not leased). But I'm assuming that you tested it with that and that it's good.

lol. that is right, @andreaskaris . I tested it using the e2e tests and did not hit any issues.

File: bindata/cloud-network-config-controller/002-rbac.yaml [hunk 1]
```diff
@@ -56,6 +56,15 @@ rules:
...
   - watch
+- apiGroups:
+  - coordination.k8s.io
+  resources:
+  - leases
+  verbs:
+  - create
+  - get
+  - list
+  - update
```
[the remark this answers is not in the document]
Comment (flavio-fernandes) [reply bindata/cloud-network-config-controller/002-rbac.yaml:67]: Ah, yeah, @msherif1234 the update is needed to acquire the lock. I actually tried w/out it and the e2e tests failed as expected. And yes, no clusterrole, only ns.

**Comment by abhat** [comment]
/approve
The aws-ovn-windows job needs to pass but any failures on that job would be unrelated to the changes here.

**Comment by flavio-fernandes** [comment]
@abhat @danwinship can we `/override e2e-aws-ovn-windows` for this pr? Or would it be better to make that test non-required until the issue with kube 1.24 for windows is completed? Or we should just wait for that and ensure this test start passing again?

**Pull request merged**